README.md

```diff
@@ -0,0 +1,130 @@
+# coursebuilder-student-dashboard
+
+This is a Course Builder module which allows students to view and manage their courses and lessons 
+in a personalised way. It keeps your lessons organized in one place and also keep track of your progress on each lesson.
+
+The student dashboard is a custom module that was developed as a standalone system i.e 
+changes to the course builder codebase was avoided where possible. The module can be integrated 
+into the Course builder without breaking anything in the application.
+
+## Requirements
+
+You'll need a system with current versions of `bash` and `git`, as well as
+`python` 2.7.
+
+## Getting started
+
+First, clone Course Builder and change directory to the Course Builder root:
+
+```sh
+git clone https://github.com/google/coursebuilder-core.git
+cd coursebuilder-core/coursebuilder
+```
+
+Course Builder provides a management script for fetching modules, so you always
+start by grabbing Course Builder, then using it to fetch the module you want to
+work with. Let's grab the student dashboard module, called `coursebuilder-student-dashboard`.
+
+Next: create a file titled install.sh. For instance, you can place the file in your desktop and copy the script below into it.
+
+```
+  #!/bin/bash
+  set -e 
+  set -o pipefail
+  sh scripts/modules.sh \
+    --targets=coursebuilder-student-dashboard@https://github.com/andela-iakande/coursebuilder-student-dashboard.git
+
+  sleep 15 #pause shell script
+
+  cat modules/coursebuilder-student-dashboard/gql.py > modules/gql/gql.py
```

Comment:
@andela-iakande Do we really intend to this? Would this not be changing GCB, which is what we are trying to avoid?

Reply:
@chege To the best of my knowledge In as much as we are implementing GraphQL with the student dashboard we will definitely have reasons to inject some codes in the GCB. In the gql.py new variables were initialized for the student dashboard, the Query object inherits more than one ObjectType. I tried the best-suggested solution which is to create the concept of abstract ObjectTypes in the student dashboard module but it required me to change more of the existing codes on the gql.py in GCB and that was why I resulted to this approach as it appears clean. I will appreciate any suggestion(s) that will make this work as you intend.

Reply:
@andela-iakande seen these https://github.com/google/coursebuilder-core/blob/0a4787dda500586a0aa382ec17d046554ec800a6/coursebuilder/modules/explorer/graphql.py and https://github.com/google/coursebuilder-core/blob/master/coursebuilder/modules/courses/graphql.py on extending anything to learn from these?

Reply:
ok, Looking into it